packages/components/src/components/button/_button.scss

```
@@ -285,6 +285,9 @@
     path:not([data-icon-path]):not([fill='none']),
   .#{$prefix}--btn--ghost.#{$prefix}--btn--icon-only .#{$prefix}--btn__icon {
     fill: $icon-01;
+    @include high-contrast-mode {
+      fill: $icon-03;
+    }
```

We can use the `icon-fill` parameter instead to ensure the fill is the same as the ButtonText forced color

https://github.com/carbon-design-system/carbon/blob/3c9fdbcc12116759c47605ad562aa4d3be78b9d5/packages/components/src/globals/scss/_helper-mixins.scss#L256-L258

```suggestion
    @include high-contrast-mode('icon-fill');
```